docs/src/rules/no-useless-escape.md

```diff
@@ -67,6 +67,42 @@ Examples of **correct** code for this rule:
 
 :::
 
+## Options
+
+This rule has an object option:
+
+* `allowedCharacters` - An array of characters that should be allowed to have unnecessary escapes in regular expressions. This is useful for characters like `-` where escaping can prevent accidental character ranges. For example, in `/[0\-]/`, the escape is technically unnecessary but helps prevent the pattern from becoming a range if another character is added later (e.g., `/[0\-9]/` vs `/[0-9]/`).
```

Comment:
I wonder if this should be `allowCharacters` to be a verb instead of a noun? Thoughts @eslint/eslint-team?

Reply:
Another thing to consider is that this rule checks string literals, template literals, and regular expression literals. Since this option applies to regular expression literals only, perhaps its name should indicate that?

Reply:
How about `allowRegexCharacters`?

Reply:
It seems we all agree about `allowRegexCharacters`.